mono/mini/debugger-agent.c

```diff
@@ -3861,10 +3995,18 @@ thread_startup (MonoProfiler *prof, uintptr_t tid)
 	if (tls) {
 		if (!tls->terminated)
 			MONO_GC_UNREGISTER_ROOT(tls->thread);
+#ifdef IL2CPP_MONO_DEBUGGER
+		il2cpp_gc_free_fixed (tls);
```

Comment:
Could we not do a VM_FREE and VM_ALLOC_FIXED like we're doing with other such macros?

Reply:
Yes, that would be possible. In general, we have tried to keep the macros for field access. Any other changes are usually inline with defines. That distinction is a bit arbitrary though.